Modules/System/SharePoint/src/helpers/SharePointRequestManager.Codeunit.al

```diff
@@ -0,0 +1,94 @@
+codeunit 9109 "SharePoint Request Manager"
+{
+    Access = Internal;
+
+    var
+        HttpClient: HttpClient;
+        Authorization: Interface "SharePoint Authorization";
+        HttpResponseInfoErr: Label '%1.\\Response Code: %2 %3', Comment = '%1 = Default Error Message ; %2 = Status Code; %3 = Reason Phrase';
+        OperationNotSuccessfulErr: Label 'An error has occurred';
+
+    procedure SetAuthorization(Auth: Interface "SharePoint Authorization")
+    begin
+        Authorization := Auth;
+    end;
+
+    procedure Get(SharePointUriBuilder: Codeunit "SharePoint Uri Builder") OperationResponse: Codeunit "SharePoint Operation Response"
+    begin
+        OperationResponse := SendRequest(PrepareRequestMsg("Http Request Type"::GET, SharePointUriBuilder));
+    end;
+
+    procedure Post(SharePointUriBuilder: Codeunit "SharePoint Uri Builder") OperationResponse: Codeunit "SharePoint Operation Response"
+    var
+        SharePointHttpContent: Codeunit "SharePoint Http Content";
+
+    begin
+        OperationResponse := SendRequest(PrepareRequestMsg("Http Request Type"::POST, SharePointUriBuilder, SharePointHttpContent));
+    end;
+
+    procedure Post(SharePointUriBuilder: Codeunit "SharePoint Uri Builder"; SharePointHttpContent: Codeunit "SharePoint Http Content") OperationResponse: Codeunit "SharePoint Operation Response"
+    begin
+        OperationResponse := SendRequest(PrepareRequestMsg("Http Request Type"::POST, SharePointUriBuilder, SharePointHttpContent));
+    end;
+
+    [NonDebuggable]
+    local procedure PrepareRequestMsg(HttpRequestType: Enum "Http Request Type"; SharePointUriBuilder: Codeunit "SharePoint Uri Builder") RequestMessage: HttpRequestMessage
+    var
+        Headers: HttpHeaders;
+    begin
+        RequestMessage.Method(Format(HttpRequestType));
+        RequestMessage.SetRequestUri(SharePointUriBuilder.GetUri());
+        RequestMessage.GetHeaders(Headers);
+        Headers.Add('Accept', 'application/json');
+    end;
+
+    [NonDebuggable]
+    local procedure PrepareRequestMsg(HttpRequestType: Enum "Http Request Type"; SharePointUriBuilder: Codeunit "SharePoint Uri Builder"; SharePointHttpContent: Codeunit "SharePoint Http Content") RequestMessage: HttpRequestMessage
+    var
+        Headers: HttpHeaders;
+        HttpContent: HttpContent;
+    begin
+        RequestMessage.Method(Format(HttpRequestType));
+        RequestMessage.SetRequestUri(SharePointUriBuilder.GetUri());
+
+        RequestMessage.GetHeaders(Headers);
+        Headers.Add('Accept', 'application/json;odata=verbose');
+
+        if SharePointHttpContent.GetContentLength() > 0 then begin
+            HttpContent := SharePointHttpContent.GetContent();
+            HttpContent.GetHeaders(Headers);
+
+            if Headers.Contains('Content-Length') then
+                Headers.Remove('Content-Length');
+            Headers.Add('Content-Length', Format(SharePointHttpContent.GetContentLength()));
```

Comment:
Just a comment: HttpClient automatically adds a content-length header based on the size of the data supplied.

Reply:
Thanks, fixed.